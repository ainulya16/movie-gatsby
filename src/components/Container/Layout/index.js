import React from 'react';
import PropTypes from 'prop-types';
import { Layout } from 'antd';

import Header from '../../Header';
import Footer from './Footer';
import ContainerWrapper from '../../Wrapper/Container';

class Container extends React.PureComponent {
  render() {
    const { children, custom } = this.props;
    return (
      <Layout>
        <Header />
        <Layout>
          <ContainerWrapper>{children}</ContainerWrapper>
          <Footer custom={custom} />
        </Layout>
      </Layout>
    );
  }
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
  custom: PropTypes.bool,
};

Container.defaultProps = {
  custom: false,
};

export default Container;
