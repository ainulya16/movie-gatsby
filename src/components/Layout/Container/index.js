import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  min-height: 100%;
  .flex-center {
    display: flex;
    flex: 1;
    justify-content: center;
    align-items: center;
  }
`;

export default Container;
