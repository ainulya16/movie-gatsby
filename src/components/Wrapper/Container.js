import styled from 'styled-components';

const Container = styled.div`
  @media (max-width: 768px) {
    margin: 5px;
  }
  /* Small devices (tablets, 768px and up) */
  @media (min-width: 768px) and (max-width: 992px) {
    margin: 10px;
  }
  /* Medium devices (desktops, 992px and up) */
  @media (min-width: 992px) and (max-width: 1200px) {
    margin: 44px;
  }
  /* Large devices (large desktops, 1200px and up) */
  @media (min-width: 1200px) {
    margin: 64px;
  }

  /* padding: 0 50px; */
  display: block;
  margin: 64px;
  @media (max-width: 768px) {
    margin: 64px 0 0 0;
  }
`;

export default Container;
