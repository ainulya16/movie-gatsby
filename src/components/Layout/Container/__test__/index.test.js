import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Container from '..';

describe('Component: Layout', () => {
  it('should be exist', () => {
    const wrapper = shallow(<Container />);
    expect(wrapper.exists()).toBe(true);
  });
  it('renders Layout default styles', () => {
    const tree = renderer
      .create(
        <Container>
          <p>Test</p>
        </Container>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
