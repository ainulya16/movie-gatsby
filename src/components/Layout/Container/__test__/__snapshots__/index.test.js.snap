// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Layout renders Layout default styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100%;
}

.c0 .flex-center {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
>
  <p>
    Test
  </p>
</div>
`;
